docs/contracts-intro.md

```diff
@@ -2,11 +2,181 @@
 
 ---8<-- 'includes/data-builder-danger-header.md'
 
-This page will introduce OpenSAFELY Contracts, explaining:
+## OpenSAFELY Contracts are a clinical data specification
 
-* what they are
-* the process by which new contracts could be proposed
-* where [details of existing contracts will be
-  found](contracts-reference.md)
+*OpenSAFELY Contracts* are an important OpenSAFELY concept that is
+introduced with Data Builder.
+
+This page describes:
+
+* what OpenSAFELY Contracts are
+* how OpenSAFELY Contracts relate to dataset definitions
+
+Each OpenSAFELY Contract is a formal specification, [described in
+code](https://github.com/opensafely-core/databuilder/blob/main/databuilder/contracts/contracts.py),
+relating to a particular clinical data domain. OpenSAFELY Contracts are
+authored by the OpenSAFELY platform developers: the Contract designs are
+informed by discussion and feedback from data providers and researchers.
+
+Researchers write dataset definitions to access data tables from
+OpenSAFELY backends. All available data tables within OpenSAFELY satisfy
+an OpenSAFELY Contract, regardless of which data provider operates the
+backend.
+
+TODO: does a table only satisfy one contract, or possibly multiple?
```

Comment:
Only one at the moment and I think that a one-to-one relationship makes sense since the marginal cost of new tables is tiny.